### PR TITLE
Problem: missing assign statedb to ctx

### PR DESCRIPTION
**Pull request by mmsqe**
Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -45,6 +45,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (statedb) [#333](https://github.com/crypto-org-chain/ethermint/pull/333) Support native action in statedb, prepare for precompiles.
 * (rpc) [#369](https://github.com/crypto-org-chain/ethermint/pull/369) Support state overrides in eth_call.
 * (precompile) [#371](https://github.com/crypto-org-chain/ethermint/pull/371) Add StateDB itself into native context for precompiles to emit evm logs directly.
+* (precompile) [#](https://github.com/crypto-org-chain/ethermint/pull/) Add missing assign statedb to ctx.
```
Comment (mmsqe) [inline CHANGELOG.md]:
```suggestion
* (precompile) [#372](https://github.com/crypto-org-chain/ethermint/pull/372) Add missing assign statedb to ctx.
```

File: x/evm/keeper/state_transition.go [hunk 2]
```diff
@@ -354,7 +354,7 @@ func (k *Keeper) ApplyMessageWithConfig(ctx sdk.Context,
 			return nil, errorsmod.Wrap(err, "failed to apply state override")
 		}
 	}
-
+	ctx = ctx.WithValue(statedb.StateDBContextKey, stateDB)
```
Comment (yihuang) [inline x/evm/keeper/state_transition.go:357]: It's already done in state db constructor.

Comment (mmsqe) [reply x/evm/keeper/state_transition.go:357]: seems still have issue, we can't modify ctx outside

Comment (yihuang) [reply x/evm/keeper/state_transition.go:357]: hooks?

Comment (yihuang) [reply x/evm/keeper/state_transition.go:357]: hooks?

Comment (yihuang) [reply x/evm/keeper/state_transition.go:357]: what issues?

Comment (yihuang) [reply x/evm/keeper/state_transition.go:357]: what issues?

Comment (mmsqe) [reply x/evm/keeper/state_transition.go:357]: we can't get stateDb [ctx](https://github.com/crypto-org-chain/cronos/blob/9cb8be65e2adf5fc392d584ebff1774d4387c841/x/cronos/keeper/evm.go#L42) without using relayer precompile?

Comment (yihuang) [reply x/evm/keeper/state_transition.go:357]: no, https://github.com/crypto-org-chain/cronos/pull/1233 don't need to change relayer precompile.

Comment (yihuang) [reply x/evm/keeper/state_transition.go:357]: no, https://github.com/crypto-org-chain/cronos/pull/1233 don't need to change relayer precompile.

Comment (mmsqe) [reply x/evm/keeper/state_transition.go:357]: no log [without](https://github.com/crypto-org-chain/cronos/pull/1233/commits/25c3feedfb8e85b2509f82a39cc0743e7c778a2d) using relayer precompile config

Comment (mmsqe) [reply x/evm/keeper/state_transition.go:357]: no log [without](https://github.com/crypto-org-chain/cronos/pull/1233/commits/25c3feedfb8e85b2509f82a39cc0743e7c778a2d) using relayer precompile config

**Pull request closed without merging**